### PR TITLE
[BACKLOG-19566] Adding an ACLProvider implementation for worker nodes that does not update the root node acl

### PR DESCRIPTION
**Pull request by pentaho-amessier**
Related PR: https://github.com/pentaho/pentaho-ee/pull/814

@pentaho/rebelalliance  @pentaho/rogueone

### CODE REVIEW

File: repository/src/main/java/org/apache/jackrabbit/core/security/authorization/acl/WorkerNodePentahoACLProvider.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.RepositoryException;
+
+/**
+ * An implementation of {@link PentahoACLProvider} that does not update the root node acl. Meant for use by the worker
+ * nodes, which do not need to update the root node acl, as the update is performed by the main pentaho server.
+ */
+public class WorkerNodePentahoACLProvider extends PentahoACLProvider {
```
Comment (pentaho-amessier) [inline repository/src/main/java/org/apache/jackrabbit/core/security/authorization/acl/WorkerNodePentahoACLProvider.java]: I'm wondering whether this class belongs in the platform or in the wn osgi feature... Not sure if it's possible to wire beans in repository.xml that aren't in the platform, will check

File: repository/src/main/java/org/apache/jackrabbit/core/security/authorization/acl/WorkerNodePentahoACLProvider.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  private Logger logger = LoggerFactory.getLogger( getClass().getName() );
+
+  @Override
+  protected void updateRootAcl( SessionImpl systemSession, ACLEditor editor ) throws RepositoryException {
```
Comment (chiltonj) [inline repository/src/main/java/org/apache/jackrabbit/core/security/authorization/acl/WorkerNodePentahoACLProvider.java]: Suggestion: add a comment explaining why we are not updating root.

Comment (pentaho-amessier) [reply repository/src/main/java/org/apache/jackrabbit/core/security/authorization/acl/WorkerNodePentahoACLProvider.java]: Is the comment at the top of the class not enough? Shall I repeat it here?

**Comment by pentaho-amessier** [comment]
Closing - will post new PRs with additional changes

**Pull request closed without merging**